packages/frontend/workspace/src/providers/index.ts

```diff
@@ -1,160 +1,129 @@
+          : createIndexedDBStorage(doc.guid),
+        []
+      );
+
+      const awarenessProviders = [
+        createBroadcastChannelAwarenessProvider(doc.guid, awareness),
+      ];
+
+      let connected = false;
+
+      return {
+        flavour: '_',
+        passive: true,
+        active: true,
+        sync() {
+          if (!connected) {
+            engine.start();
+
+            for (const provider of awarenessProviders) {
+              provider.connect();
+            }
+            connected = true;
           }
-          _resolve();
-        })
-        .catch(_reject);
+        },
+        get whenReady() {
+          return engine.waitForLoadedRootDoc();
+        },
+        connect() {
+          // TODO: actually connect
+        },
+        disconnect() {
+          // TODO: actually disconnect
+        },
+        get connected() {
+          return connected;
+        },
+        engine,
+      };
     },
-  };
-};
-
-export {
-  createAffineSocketIOProvider,
-  createBroadcastChannelProvider,
-  createIndexedDBBackgroundProvider,
-  createIndexedDBDownloadProvider,
-  createSQLiteDBDownloadProvider,
-  createSQLiteProvider,
-  downloadBinaryFromCloud,
-};
-
-export const createLocalProviders = (): DocProviderCreator[] => {
-  const providers = [
-    createIndexedDBBackgroundProvider,
-    createIndexedDBDownloadProvider,
-  ] as DocProviderCreator[];
-
-  if (runtimeConfig.enableBroadcastChannelProvider) {
-    providers.push(createBroadcastChannelProvider);
-  }
-
-  if (environment.isDesktop && runtimeConfig.enableSQLiteProvider) {
-    providers.push(createSQLiteProvider, createSQLiteDBDownloadProvider);
-  }
-
-  return providers;
+  ];
 };
 
 export const createAffineProviders = (): DocProviderCreator[] => {
-  return (
-    [
-      ...createLocalProviders(),
-      runtimeConfig.enableCloud && createAffineSocketIOProvider,
-      runtimeConfig.enableCloud && createMergeCloudSnapshotProvider,
-    ] as DocProviderCreator[]
-  ).filter(v => Boolean(v));
+  return [
+    (_, doc, { awareness }) => {
+      const engine = new SyncEngine(
+        doc,
+        environment.isDesktop
+          ? createSQLiteStorage(doc.guid)
+          : createIndexedDBStorage(doc.guid),
+        [createAffineStorage(doc.guid)]
+      );
+
+      const awarenessProviders = [
+        createBroadcastChannelAwarenessProvider(doc.guid, awareness),
+        createAffineAwarenessProvider(doc.guid, awareness),
+      ];
+
+      let connected = false;
+
+      return {
+        flavour: '_',
+        passive: true,
+        active: true,
+        sync() {
+          if (!connected) {
+            engine.start();
+
+            for (const provider of awarenessProviders) {
+              provider.connect();
+            }
+            connected = true;
+          }
+        },
+        get whenReady() {
+          console.log('!!!!!!!!!!!!!!!');
+          return engine.waitForLoadedRootDoc();
+        },
+        connect() {
```

Comment:
looks like we don't need the provider to be PassiveProvider

Reply:
Just leave this here and tell us this needs to be refactored.